logger/redact.go

```diff
@@ -146,18 +134,16 @@ func RedactFormURLEncoded(form string, replacement string, keys [][]string) stri
 	if err != nil {
 		return form
 	}
-	bytes, err := json.Marshal(r)
-	if err != nil {
-		return form
-	}
-	redactedBody := RedactJSON(string(bytes), replacement, keys)
 
-	// Unmarshal the redacted JSON to a url.Values
-	var redactedForm url.Values
-	err = json.Unmarshal([]byte(redactedBody), &redactedForm)
-	if err != nil {
-		return form
+	for _, key := range keys {
+		if len(key) >= 1 {
+			if values, exists := r[key[0]]; exists {
```

Comment:
just wondering if we only use the first key, can't we use map instead?

Reply:
Refactored to make things more readable, and also changed the function call to take in `[]string` instead - 1048e35. Also added `ExampleRedactFormURLEncoded`.